packages/sdk/src/lib/Contracts.ts

```diff
@@ -59,20 +59,31 @@ export class Contract {
    * @param signingKeyPair key pair to sign the deploy
    */
   public deploy(
-    args: Deploy.Arg[],
+    args: RuntimeArgs,
     paymentAmount: bigint,
     accountPublicKeyHash: ByteArray,
     signingKeyPair: nacl.SignKeyPair
   ): Deploy {
-    const deploy = makeDeploy(
-      args,
-      ContractType.WASM,
+    const session = new DeployUtil.ModuleBytes(
       this.sessionWasm,
+      args.toBytes()
+    );
+    const paymentArgs = RuntimeArgs.fromMap({
+      amount: CLValue.fromU512(paymentAmount.toString())
+    });
+
+    const payment = new DeployUtil.ModuleBytes(
       this.paymentWasm,
-      paymentAmount,
-      accountPublicKeyHash
+      paymentArgs.toBytes()
     );
-    return signDeploy(deploy, signingKeyPair);
+
+    const deploy = DeployUtil.makeDeploy(
+      session,
+      payment,
+      accountPublicKeyHash,
+      'casper-net-1'
```

Comment:
That should not be hardcoded. Should come from the config.

Reply:
Done.